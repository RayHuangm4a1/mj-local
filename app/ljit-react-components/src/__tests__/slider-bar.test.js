import React from 'react';
import { shallow, mount, } from 'enzyme';
import SliderBar from '../components/slider-bar';

jest.mock('antd/lib/slider');

describe('Slider Bar', () => {
	it('handle default props', () => {
		const {
			value,
			min,
			max,
			step,
			disabled,
			suffix,
			onChange,
		} = SliderBar.defaultProps;

		expect(value).toEqual(0);
		expect(min).toBe(0);
		expect(max).toBe(100);
		expect(step).toBe(1);
		expect(disabled).toEqual(false);
		expect(suffix).toBe('单位');
		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const value = 10;
		const min = 0;
		const max = 1000;
		const wrapper = shallow(
			<SliderBar
				min={min}
				max={max}
				value={value}
				onChange={() => {}}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-slider-bar', () => {
		const wrapper = shallow(<SliderBar />);

		expect(wrapper.hasClass('ljit-slider-bar')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<SliderBar className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const value = 10;
		const max = 100;
		const min = 1;
		const step = 0.1;
		const suffix = 'mock-suffix';
		const className = 'mock-class';
		const disabled = false;
		const onChange = () => {};

		const wrapper = mount(
			<SliderBar
				value={value}
				max={max}
				min={min}
				step={step}
				suffix={suffix}
				className={className}
				disabled={disabled}
				onChange={onChange}
			/>
		);

		expect(wrapper.props().value).toBe(value);
		expect(wrapper.props().max).toBe(max);
		expect(wrapper.props().min).toBe(min);
		expect(wrapper.props().step).toBe(step);
		expect(wrapper.props().suffix).toBe(suffix);
		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().disabled).toEqual(disabled);
		expect(wrapper.props().onChange).toEqual(onChange);
	});
});
