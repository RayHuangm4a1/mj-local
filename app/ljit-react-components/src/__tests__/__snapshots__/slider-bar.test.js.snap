// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Bar should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderBar
    disabled={false}
    max={1000}
    min={0}
    onChange={[Function]}
    step={1}
    suffix="单位"
    value={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Slider
          className="ljit-slider-bar__slider"
          disabled={false}
          max={1000}
          min={0}
          onChange={[Function]}
          step={1}
          tipFormatter={[Function]}
          value={10}
        />,
        <InputNumber
          className="ljit-slider-bar__number"
          disabled={false}
          formatType=""
          max={1000}
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          step={1}
          value={10}
        />,
        <div
          className="ljit-slider-bar__suffix"
        >
          单位
        </div>,
      ],
      "className": "ljit-slider-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "ljit-slider-bar__slider",
          "disabled": false,
          "max": 1000,
          "min": 0,
          "onChange": [Function],
          "step": 1,
          "tipFormatter": [Function],
          "value": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "ljit-slider-bar__number",
          "disabled": false,
          "formatType": "",
          "max": 1000,
          "min": 0,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "readOnly": false,
          "step": 1,
          "value": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "单位",
          "className": "ljit-slider-bar__suffix",
        },
        "ref": null,
        "rendered": "单位",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Slider
            className="ljit-slider-bar__slider"
            disabled={false}
            max={1000}
            min={0}
            onChange={[Function]}
            step={1}
            tipFormatter={[Function]}
            value={10}
          />,
          <InputNumber
            className="ljit-slider-bar__number"
            disabled={false}
            formatType=""
            max={1000}
            min={0}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={false}
            step={1}
            value={10}
          />,
          <div
            className="ljit-slider-bar__suffix"
          >
            单位
          </div>,
        ],
        "className": "ljit-slider-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "ljit-slider-bar__slider",
            "disabled": false,
            "max": 1000,
            "min": 0,
            "onChange": [Function],
            "step": 1,
            "tipFormatter": [Function],
            "value": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "ljit-slider-bar__number",
            "disabled": false,
            "formatType": "",
            "max": 1000,
            "min": 0,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "readOnly": false,
            "step": 1,
            "value": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "单位",
            "className": "ljit-slider-bar__suffix",
          },
          "ref": null,
          "rendered": "单位",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
